cpp/bench/ann/src/cuvs/cuvs_benchmark.cu

```diff
@@ -113,6 +174,29 @@ auto create_search_param(const std::string& algo_name, const nlohmann::json& con
     return param;
   }
 #endif
+#ifdef CUVS_ANN_BENCH_USE_CUVS_MG
+  if (algo_name == "raft_mg_ivf_flat" || algo_name == "cuvs_mg_ivf_flat") {
+    auto param =
+      std::make_unique<typename cuvs::bench::cuvs_mg_ivf_flat<T, int64_t>::search_param>();
+    parse_search_param<T, int64_t>(conf, *param);
+    add_merge_mode(&param->merge_mode, conf);
+    return param;
+  }
+
+  if (algo_name == "raft_mg_ivf_pq" || algo_name == "cuvs_mg_ivf_pq") {
+    auto param = std::make_unique<typename cuvs::bench::cuvs_mg_ivf_pq<T, int64_t>::search_param>();
+    parse_search_param<T, int64_t>(conf, *param);
+    add_merge_mode(&param->merge_mode, conf);
+    return param;
+  }
+
+  if (algo_name == "raft_mg_cagra" || algo_name == "cuvs_mg_cagra") {
+    auto param = std::make_unique<typename cuvs::bench::cuvs_mg_cagra<T, uint32_t>::search_param>();
+    parse_search_param<T, uint32_t>(conf, *param);
```

Comment:
Here (and above in the mg cases) can't we just parse the single GPU param struct, and construct the parallel param struct from that?

Reply:
The MG bench struct inherits from the SG bench struct. Therefore, in the MG case the full MG struct should be allocated which is not necessary for the SG case. The main parameters (the SG ones) are filled with the `parse_search_param` function, and the `add_merge_mode` adds the specifically MG parameter. In my opinion, it is relatively clean, but I might be missing some bits to improve the code even further though.

Reply:
I agree. Except to fill the main parameters, I would not have a specific overload of `parse_search_param` for the mg case (see my comment [here](https://github.com/rapidsai/cuvs/pull/231/files#r1777733285)). Why does the cagra param reader have separate variants for the SG and MG bench case, while other methods do not?

Reply:
I wrote this code quite some time ago. Actually, for CAGRA bench, the build parameters for the MG bench did not inherit from the SG bench. I could fix this with a number of changes.